doc/developer/assets/crate-diagram.svg

```diff
@@ -0,0 +1,1790 @@
+</a>
+</g>
+</g>
+<!-- s3&#45;datagen&#45;&gt;ore -->
+<g id="edge142" class="edge">
+<title>s3&#45;datagen&#45;&gt;ore</title>
+<g id="a_edge142"><a xlink:href="none" xlink:title="s3&#45;datagen &#45;&gt; ore">
+<path fill="none" stroke="black" d="M2773.33,-677.24C2788.83,-672.1 2808.56,-666.05 2826.5,-662 2958.07,-632.34 3106.5,-679.87 3106.5,-545 3106.5,-545 3106.5,-545 3106.5,-177 3106.5,-134.16 3104.82,-112.24 3069.5,-88 2985.18,-30.11 1367.92,-33.82 1099.96,-34.84"/>
+<polygon fill="black" stroke="black" points="1099.59,-31.34 1089.61,-34.88 1099.62,-38.34 1099.59,-31.34"/>
+</a>
+</g>
+</g>
+<!-- smith&#45;fuzz -->
+<g id="node41" class="node">
+<title>smith&#45;fuzz</title>
+<g id="a_node41"><a xlink:title="Fuzzer intergration using Smith to generate Materialize inputs">
+<polygon fill="none" stroke="red" points="1910.5,-706 1841.63,-688 1910.5,-670 1979.38,-688 1910.5,-706"/>
+<polyline fill="none" stroke="red" points="1853.24,-691.03 1853.24,-684.97 "/>
+<polyline fill="none" stroke="red" points="1898.89,-673.03 1922.11,-673.03 "/>
+<polyline fill="none" stroke="red" points="1967.77,-684.97 1967.77,-691.03 "/>
+<polyline fill="none" stroke="red" points="1922.11,-702.97 1898.89,-702.97 "/>
+<text text-anchor="middle" x="1910.5" y="-684.3" font-family="Times,serif" font-size="14.00">smith&#45;fuzz</text>
+</a>
+</g>
+</g>
+<!-- smith&#45;fuzz&#45;&gt;test&#45;util -->
+<g id="edge143" class="edge">
+<title>smith&#45;fuzz&#45;&gt;test&#45;util</title>
+<g id="a_edge143"><a xlink:href="none" xlink:title="smith&#45;fuzz &#45;&gt; test&#45;util">
+<path fill="none" stroke="black" d="M1937.99,-676.97C1952.79,-671.88 1971.47,-665.95 1988.5,-662 2079.53,-640.89 2188.15,-627.73 2248.73,-621.43"/>
+<polygon fill="black" stroke="black" points="2249.16,-624.91 2258.75,-620.41 2248.45,-617.95 2249.16,-624.91"/>
+</a>
+</g>
+</g>
+<!-- test&#45;util&#45;&gt;kafka&#45;util -->
+<g id="edge144" class="edge">
+<title>test&#45;util&#45;&gt;kafka&#45;util</title>
+<g id="a_edge144"><a xlink:href="none" xlink:title="test&#45;util &#45;&gt; kafka&#45;util">
+<path fill="none" stroke="black" d="M2257.8,-615.21C2147.6,-613.99 1820.77,-597.06 1613.5,-446 1554.8,-403.21 1587.45,-348.04 1527.5,-307 1490.64,-281.77 1367.86,-266.04 1296.04,-258.77"/>
+<polygon fill="black" stroke="black" points="1296.13,-255.27 1285.83,-257.76 1295.44,-262.23 1296.13,-255.27"/>
+</a>
+</g>
+</g>
+<!-- test&#45;util&#45;&gt;ore -->
+<g id="edge145" class="edge">
+<title>test&#45;util&#45;&gt;ore</title>
+<g id="a_edge145"><a xlink:href="none" xlink:title="test&#45;util &#45;&gt; ore">
+<path fill="none" stroke="black" d="M2332.8,-612.77C2496.66,-602.49 3144.5,-556.55 3144.5,-473 3144.5,-473 3144.5,-473 3144.5,-177 3144.5,-136.03 3152.7,-113.33 3120.5,-88 3079.18,-55.49 1374.38,-37.96 1099.69,-35.34"/>
+<polygon fill="black" stroke="black" points="1099.49,-31.84 1089.45,-35.25 1099.42,-38.84 1099.49,-31.84"/>
+</a>
+</g>
+</g>
+</g>
+</svg>
```

Comment:
What enforces that this diagram will stay up to date?

Reply:
right now code review and requiring the lint on the .gv file. I can change the cli api to always only write both, though, which should make it both easier to use and harder to screw up.